casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -166,7 +165,7 @@ object Validate {
       true.pure[F]
     }
 
-  def version[F[_]: Applicative: Log](b: BlockMessage, version: Long): F[Boolean] = {
+  def version[F[_]: Monad: Log](b: BlockMessage, version: Long): F[Boolean] = {
```

Comment:
Shouldn't this be `FlatMap` as it is in your change to `PacketHandler`?

Reply:
this me field needs applicative otherwise